cli/src/lib.rs

```diff
@@ -3071,22 +3089,40 @@ fn run_test_suite(
     skip_local_validator: bool,
     skip_deploy: bool,
     detach: bool,
+    validator_type: ValidatorType,
     test_validator: &Option<TestValidator>,
     scripts: &ScriptsConfig,
     extra_args: &[String],
+    surfpool_config: &Option<SurfpoolConfig>,
 ) -> Result<()> {
     println!("\nRunning test suite: {:#?}\n", test_suite_path.as_ref());
-    // Start local test validator, if needed.
     let mut validator_handle = None;
-    if is_localnet && (!skip_local_validator) {
-        let flags = match skip_deploy {
-            true => None,
-            false => Some(validator_flags(cfg, test_validator)?),
-        };
-        validator_handle = Some(start_test_validator(cfg, test_validator, flags, true)?);
+    match validator_type {
+        ValidatorType::Surfpool => {
+            if is_localnet && (!skip_local_validator) {
+                let flags = match skip_deploy {
+                    true => None,
+                    false => Some(surfpool_flags(cfg, surfpool_config, false)?),
+                };
+                validator_handle = Some(start_surfpool_validator(flags, surfpool_config)?);
+            }
+        }
```

Comment:
You could DRY up this code by checking `if is_localnet && (!skip_local_validator)` only once, then matching on validator type, rather than matching on the type then checking that same condition twice.